### PR TITLE
feat(fetch): add support for HTTP2 pseudo headers

### PR DESCRIPTION
**Pull request by pcattori**

### CODE REVIEW

**Review by mjackson — APPROVED** [review]

**Comment by brophdawg11** [comment]
Want to to add a unit test alongside the `should reject illegal header` test (packages/fetch/test/headers.js`) file to assert these are ok?

File: packages/fetch/src/headers.js [hunk 1]
```diff
@@ -8,21 +8,19 @@ import {types} from 'util';
 import http from 'http';
 import { isIterable } from './utils/is.js'
 
-const validators = /** @type {{validateHeaderName?:(name:string) => any, validateHeaderValue?:(name:string, value:string) => any}} */
-(http)
+/** @type {{validateHeaderValue?:(name:string, value:string) => any}} */
+const validators = (http)
 
-const validateHeaderName = typeof validators.validateHeaderName === 'function' ?
-	validators.validateHeaderName :
```
Comment (brophdawg11) [inline packages/fetch/src/headers.js:15]: Do we know exactly what this used to do?  Any chance we start inadvertently rejecting headers in the wild that use to be allowed by this?

Comment (pcattori) [reply packages/fetch/src/headers.js:15]: The `http`-based `validateHeaderName` (which is what `validators.validateHeaderName` is) looks the same as ours. The only difference now is the `:`

https://github.com/nodejs/node/blob/main/lib/_http_outgoing.js#L664-L668
https://github.com/nodejs/node/blob/main/lib/_http_common.js#L206-L214

**Comment by pcattori** [comment]
@brophdawg11 good callout, just added a test. Fails before this change, passes now.

**Pull request merged**